File: WorkLogService/app/schemas/work_log.py
from datetime import date, datetime
from typing import Optional

from pydantic import BaseModel, ConfigDict


class WorkLog(BaseModel):
    model_config = ConfigDict(from_attributes=True)
    id: int
    user_id: int
    description: str
    llm_Description: Optional[str] = None
    project: Optional[str] = None
    date: date
    created_at: datetime
    updated_at: datetime
    status: str


class WorkLogCreate(BaseModel):
    user_id: int
    description: str
    project: Optional[str] = None
    # Дата в формате ISO (YYYY-MM-DD), если не указана
    date: Optional[str] = None
    status: Optional[str] = None


class PaginatedWorkLogs(BaseModel):
    items: list[WorkLog]
    total: int
    page: int
    page_size: int

File: Frontend-React/src/types/worklog.ts
export type WorkLogItem = {
  id: number
  user_id: number
  description: string
  llm_Description: string | null
  project: string | null
  date: string
  created_at: string
  updated_at: string
  status: string
}

export type PaginatedWorkLogs = {
  items: WorkLogItem[]
  total: number
  page: number
  page_size: number
}
